bindings/experimental/distrdf/python/DistRDF/Profiling/Base.py

```diff
@@ -0,0 +1,27 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+class ProfilingData(ABC):
+    """
+    Base class for the collection of any type of profiling data
+    """
+
+    @abstractmethod
+    def Merge(self, other:"ProfilingData"):
+        """
+        Profiled data must provide a method to merge two instances of the same type.
+        """
+        raise NotImplementedError
+
+class Visualization:
+    """
+    Base class that represents visualizations for profiling data
+    """
+
+    @property
+    def Decorator(self) -> Callable:
+        return lambda func: func
+
+    @property
+    def Client_task(self) -> Callable:
```

Comment:
I think client task is too generic of a name, there are many "tasks" that the client needs to do: setting up the computation graph, handling returned values after distributed execution etc.

Maybe this can be called `produce_visualization` or something similar?